xds/src/main/java/io/grpc/xds/RandomWeightedPicker.java

```diff
@@ -116,4 +116,17 @@ public String toString() {
         .add("totalWeight", totalWeight)
         .toString();
   }
+
+  /** Factory that creates a SubchannelPicker for a given list of weighted child pickers. */
+  interface WeightedPickerFactory {
```

Comment:
Maybe we do not need both `RandomWeightedPicker` and `WeightedPickerFactory` (I feel like we are introducing many classes/interfaces that do not seem to be necessary). 

```java
final class WeightedRamdomPicker {  // or some better class name
   private final Random rand;
   WeightedRamdonPicker(Random rand) {
      this.rand = rand;
   }
   SubchannelPicker picker(List<WeightedChildPicker> childPickers) {  // or some better method name
      // compute total weight for the list of child pickers, randomly choose one.
   }
}
```

Reply:
This comment is outdated.